### PR TITLE
Document CIP-20/30/31 precompiles

### PR DESCRIPTION
**Pull request by karlb**
The documentation to go along with https://github.com/celo-org/celo-monorepo/pull/10240.

Direct link to rendered paged: https://deploy-preview-872--celo-docs.netlify.app/protocol/precompiles

### CODE REVIEW

**Comment by karlb** [comment]
The new page works fine, but apparently there is some problem with the change to add it to the sidebar. I can't see all the logs (limited permissions?), so I'll wait for one of the reviewers to tell me what's wrong with that tiny change.

**Comment by marekolszewski** [comment]
This is a great addition! I wonder if it's worth having a more explicit example around BLS12-381 to help with SEO for folks looking for such precompiles. @kobigurk Perhaps you can think of a good example?

**Comment by kobigurk** [comment]
BLS signature verification would be great! I'll take a look

**Comment by martinvol** [comment]
Hey @karlb

Will this close this issue fully or not all precompiles get documented?

https://github.com/celo-org/celo-monorepo/issues/10163

**Comment by karlb** [comment]
> Will this close this issue fully or not all precompiles get documented?
> 
> https://github.com/celo-org/celo-monorepo/issues/10163

There certainly is potential for more documentation, but I'm not how much we want to add here. I just wanted to make sure there is some documentation at all, so the precompiles won't be missed when skimming the docs.

**Comment by karlb** [comment]
If there is anything I can do to get this merged, let me know. I think I can be useful in its current state and we should not wait for more complete precompile documentation before merging.

File: docs/protocol/precompiles.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+These wrappers can be obtained from [celo-monorepo's common contracts](https://github.com/celo-org/celo-monorepo/tree/master/packages/protocol/contracts/common). The Solidity libraries are
+
+* CIP20Lib for CIP-20 hash functions
+* B12_377Lib (in B12.sol) for CIP-30/BLS12-377 curve operations
+* B12_381Lib (in B12.sol) for CIP-31/BLS12-381 curve operations
+
+## Usage Example
+
+If you would like to use the `sha3_256` hash function, just include the `CIP20Lib` and attach its functions to the `bytes` with `using CIP20Lib for bytes`. Then you can call the hash functions like methods on you `bytes` typed variables.
```
Comment (carterqw2) [inline docs/protocol/precompiles.md:31]: Typo in `on you`, probably should be `on your`.

**Review by carterqw2 — APPROVED** [review]

**Pull request merged**